lib/commands/connect/mapping-state.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+var api = require('./shared.js');
+var cli = require('heroku-cli-util');
+var co  = require('co');
+
+module.exports = {
+  topic: 'connect',
+  command: 'mapping:state',
+  description: 'return a mapping state',
+  help: 'return a mapping state',
+  args: [ {name: 'mapping'}],
+  flags: [
+    {name: 'resource', char: 'r', description: 'specific connection resource name', hasValue: true}
+  ],
+  needsApp: true,
+  needsAuth: true,
+  run: cli.command(co.wrap(function* (context, heroku) {
```

Comment:
co and generators are recommended in [Heroku's toolbelt plugin development docs](https://devcenter.heroku.com/articles/developing-toolbelt-plug-ins#using-generators-with-co), and they've worked fine for me so far.